modules/hal_nordic/nrfx/nrfx_config.h

```diff
@@ -1,1163 +1,100 @@
-#define NRFX_IPCTx_CHANNELS_SINGLE_VAR_NAME_BY_INST_NUM(inst_num) \
-	COND_CODE_1(IS_EMPTY(inst_num), \
-		(NRFX_CONFIG_IPCT_CHANNELS_ENTRY_NAME(NRFX_CONFIG_IPCT_LOCAL_NODE)), \
-		(NRFX_CONFIG_IPCT_CHANNELS_ENTRY_NAME(DT_NODELABEL(_CONCAT(ipct, inst_num)))))
-
-/* Variables entries generation. */
-#define NRFX_CONFIG_IPCT_CHANNELS_ENTRY(node_id) \
-	static nrfx_atomic_t NRFX_CONFIG_IPCT_CHANNELS_ENTRY_NAME(node_id) \
-		__attribute__((used)) = \
-		NRFX_CONFIG_IPCT_MASK_DT(node_id);
-#define NRFX_INTERCONNECT_IPCT_LOCAL_DEFINE \
-	DT_FOREACH_STATUS_OKAY(nordic_nrf_ipct_local, NRFX_CONFIG_IPCT_CHANNELS_ENTRY)
-#define NRFX_INTERCONNECT_IPCT_GLOBAL_DEFINE \
-	DT_FOREACH_STATUS_OKAY(nordic_nrf_ipct_global, NRFX_CONFIG_IPCT_CHANNELS_ENTRY)
-#endif /* DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_ipct_global) || ... */
+/* Define nrfx configuration based on Zephyrs KConfigs. */
+#include "nrfx_kconfig.h"
 
-#include <nrfx_config_common.h>
+/* Use defaults for undefined symbols. */
+#include <templates/nrfx_config_common.h>
 #if defined(NRF51)
-    #include <nrfx_config_nrf51.h>
+    #include <templates/nrfx_config_nrf51.h>
 #elif defined(NRF52805_XXAA)
-    #include <nrfx_config_nrf52805.h>
+    #include <templates/nrfx_config_nrf52805.h>
 #elif defined(NRF52810_XXAA)
-    #include <nrfx_config_nrf52810.h>
+    #include <templates/nrfx_config_nrf52810.h>
 #elif defined(NRF52811_XXAA)
-    #include <nrfx_config_nrf52811.h>
+    #include <templates/nrfx_config_nrf52811.h>
 #elif defined(NRF52820_XXAA)
-    #include <nrfx_config_nrf52820.h>
+    #include <templates/nrfx_config_nrf52820.h>
 #elif defined(NRF52832_XXAA) || defined (NRF52832_XXAB)
-    #include <nrfx_config_nrf52832.h>
+    #include <templates/nrfx_config_nrf52832.h>
 #elif defined(NRF52833_XXAA)
-    #include <nrfx_config_nrf52833.h>
+    #include <templates/nrfx_config_nrf52833.h>
 #elif defined(NRF52840_XXAA)
-    #include <nrfx_config_nrf52840.h>
+    #include <templates/nrfx_config_nrf52840.h>
 #elif defined(NRF5340_XXAA_APPLICATION)
-    #include <nrfx_config_nrf5340_application.h>
+    #include <templates/nrfx_config_nrf5340_application.h>
 #elif defined(NRF5340_XXAA_NETWORK)
-    #include <nrfx_config_nrf5340_network.h>
+    #include <templates/nrfx_config_nrf5340_network.h>
 #elif defined(NRF54H20_XXAA) && defined(NRF_APPLICATION)
-    #include <nrfx_config_nrf54h20_application.h>
+    #include <templates/nrfx_config_nrf54h20_application.h>
 #elif defined(NRF54H20_XXAA) && defined(NRF_RADIOCORE)
-    #include <nrfx_config_nrf54h20_radiocore.h>
+    #include <templates/nrfx_config_nrf54h20_radiocore.h>
 #elif defined(NRF54H20_XXAA) && defined(NRF_PPR)
-    #include <nrfx_config_nrf54h20_ppr.h>
+    #include <templates/nrfx_config_nrf54h20_ppr.h>
 #elif defined(NRF54H20_XXAA) && defined(NRF_FLPR)
-    #include <nrfx_config_nrf54h20_flpr.h>
+    #include <templates/nrfx_config_nrf54h20_flpr.h>
+#elif defined(NRF54H20_ENGA_XXAA) && defined(NRF_APPLICATION)
```

Comment:
Do we need to add this stuff again? It is no longer used in Zephyr.

Reply:
My intention is to keep this file in sync with this one:
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hal_nordic/nrfx/nrfx_config.h
Next nrfx release should remove those unused lines, it can be aligned then.

Reply:
You mean this one: https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/templates/nrfx_config.h, right? It still need some editing anyway. It's not just simple copying of the template, so I thought this would not be that big effort to not bring back this unnecessary stuff. But I won't insist on this. It's not a big deal.